classes/village/VillageApiPresenter.php

```diff
@@ -161,6 +161,84 @@ function (WarLogDto $war_log) {
+                        "relation_type" => VillageRelation::RELATION_LABEL[$warRecordDto->village_relation->relation_type],
+                        "relation_name" => $warRecordDto->village_relation->relation_name,
+                        "relation_start" => date("n/j/Y", $warRecordDto->village_relation->relation_start),
+                        "relation_end" => $warRecordDto->village_relation->relation_end ? date("n/j/Y", $warRecordDto->village_relation->relation_end) : null,
+                    ],
+                    "attacker_war_log" => [
+                        "log_id" => $warRecordDto->attacker_war_log->log_id,
+                        "log_type" => $warRecordDto->attacker_war_log->log_type,
+                        "village_id" => $warRecordDto->attacker_war_log->village_id,
+                        "village_name" => $warRecordDto->attacker_war_log->village_name,
+                        "relation_id" => $warRecordDto->attacker_war_log->relation_id,
+                        "infiltrate_count" => $warRecordDto->attacker_war_log->infiltrate_count,
+                        "reinforce_count" => $warRecordDto->attacker_war_log->reinforce_count,
+                        "raid_count" => $warRecordDto->attacker_war_log->raid_count,
+                        "loot_count" => $warRecordDto->attacker_war_log->loot_count,
+                        "damage_dealt" => $warRecordDto->attacker_war_log->damage_dealt,
+                        "damage_healed" => $warRecordDto->attacker_war_log->damage_healed,
+                        "defense_gained" => $warRecordDto->attacker_war_log->defense_gained,
+                        "defense_reduced" => $warRecordDto->attacker_war_log->defense_reduced,
+                        "resources_stolen" => $warRecordDto->attacker_war_log->resources_stolen,
+                        "resources_claimed" => $warRecordDto->attacker_war_log->resources_claimed,
+                        "patrols_defeated" => $warRecordDto->attacker_war_log->patrols_defeated,
+                        "regions_captured" => $warRecordDto->attacker_war_log->regions_captured,
+                        "villages_captured" => $warRecordDto->attacker_war_log->villages_captured,
+                        "pvp_wins" => $warRecordDto->attacker_war_log->pvp_wins,
+                        "points_gained" => $warRecordDto->attacker_war_log->points_gained,
+                        "stability_gained" => $warRecordDto->attacker_war_log->stability_gained,
+                        "stability_reduced" => $warRecordDto->attacker_war_log->stability_reduced,
+                        "war_score" => $warRecordDto->attacker_war_log->war_score,
+                        "objective_score" => $warRecordDto->attacker_war_log->objective_score,
+                        "resource_score" => $warRecordDto->attacker_war_log->resource_score,
+                        "battle_score" => $warRecordDto->attacker_war_log->battle_score,
+                    ],
```

Comment:
This is a great candidate for extracting to its own method, could then be reused for the defender war log below as well
```suggestion
                    "attacker_war_log" => VillageApiPresenter::warLogResponse($warRecordDto->attacker_war_log),
```

Reply:
Yup makes sense.